pkg/kritis/cryptolib/pgp_e2e_test.go

```diff
@@ -0,0 +1,210 @@
+H0hk9G/wCgBsqd0aVqIyMkrISUDjKAVu4yIdjfp826hh9hKqdhG29sGLA47XraY2
+DrwFJyguU+TJ4nPHVo9xrwYAuSE66Js+JTdMlSJf6l0v7zsp44Q5STLN39qDHSSP
+DdU6LuP4bzD5raFV8Zaqvez0F4xkpaeMl+4SdOvm23t5641ZU8/j3FZccfLI3rCS
+VLpxTMEH0IWjKYBGA66QCsp+N8Nz+P4TmXuapgeKhbR2po2dO4jTonhog/AEKxho
+JpSwWmU4Am4jG019jhIPyetj89QZkF9dSiIm2tjYJUvH9rir0Hc3AEgSTkTVmWtQ
+AolXcGfBmOwqHPQv3KKlQmja1nWJAbYEGAEKACAWIQRY/LO/rMcX/yPnyf4EQjWl
+54+HTAUCXtkETgIbDAAKCRAEQjWl54+HTNDSDACHUgc7C/xjYnPbj9HrEbxLfTL1
+IZzlPNt61/aUiUc7C+b64zN9bHttkG0rRgsW7MRJdaqkak6Lj2NAk65WID4vPaCm
+1TjzYTgOAA9r3JX64rICltUZkuCkPBxaHIoP/Wyjet1EQaN1n0whmIkIqVLODuu7
+tsMlT/GLKkYBWQ/GgaEZTio9Tjwhmhv40rLmGollXEtgzqw+JMyLU6vQ8GAieiAO
+YhVnPCyPZz1SauNbXU4kerrDwCO/+ErjaEKRAT4kjblwcguVDRCw8w2XHs2r++Wm
+UWsZLSsor5TQa+/dUxbxZyLt148PGO5T3tsNj7SYK5twdJA38Vp1O+WQk9cHdsMp
+V9Phm2bQrsMgDrfa1qhUKqhwjVxxGYSv8B4a4tYnGY0YkCA1LBulMUgVR58FRvKh
+Mv5wiQifJ4AIyZpGowVw9SLAFfXgdtBB5c8I1QE0P63jyfpAPdsbMMf2Ob5e9HPz
+K04oF0MR3F1FP5+mXZmjpEQxotoud5VYlWsFqWg=
+=szUT
+-----END PGP PRIVATE KEY BLOCK-----`
+
+const e2eSignature = `-----BEGIN PGP MESSAGE-----
+
+owGbwMvMwMXI4mS69Hl/uw/jWkbFJKFUo9SQ1OKSgMTKnPzEFL2SipK4m+yTqrk4
+lZKLMksykxNzlKwUgFxOpcyU1LySzJJKKF9BQSklPzk7tUi3KDUttSg1LzkVKKOU
+nlykl5mvn5aYnaqfmFOQmZeqBFRcqwM2ITcxPRVDe25iXmYa0Am6KZnpQApkSHFG
+opGpmVWSQVKqZWJyskGyRXKikaFJmplJWqJJqkEqkJ2YlGyQlGZsYpRqkWiSlGxk
+nGSeamFilJhkZJiclmRqkGiWmoiwuaSyAOw69/z89JxUheSc/NIUhaTMvMTSkowq
+heT8vJJEoEuLFIoz0/MSS0qLQI5WUKjlqu1kPMXCwMjFYCKmyBLxZ/P+NcfF/ys/
+P/kPFoisTKDgEpMpSC+IBwalQ3JRZUFJfk5mkl5yfi4DF6cATOH+Rdz/zI+0ubRe
+nbb3dNe8ifP736eclvwiJ3fMO5Blb+YKU79FrkpX/kSuDtimrObP2vdwHfPlFxv/
+nmpmlkjmWRyun3KB78Devq1GG/SvWmmdr7i+LfXdmr70jo2isZPbbJ4nHiv/xNh1
+Y5PEghvSzqEMgYeWLDa781Bn8z7LzJnL/7xvq7+fdf/PgmtPvW6fPvrLpHveTSN7
+KzfLwEYnsekHj9+1mPomz0woKbYiZvHi/ToFhR43/GwXLXfTEnlz//iBrl+iyyUX
+cFs+S9O6O/mCfgxHw8VP5yw/X/0ZJmeY3B1mE2qV9YO7LvHTMVuJ99O78z43HtWX
+2dCwV+Aen1qPyRomgWUzq55vs1oz41KV4przUuV3rny5tvHmzM+83ordyiZSjwpN
+OyU2MES8OPdZLmG2/enlTP2TF63Y2fXS78H91NklnjIrGI047quvut1RZq+o/F8/
+pOG2u9GF9/9YXp9otObYveiz69LNXxNW7WSZkCT/2bV1GcNXk4jzXUcMNsvOPjpT
+TOjPfd+FW+LmXZnGbVh2vwMA
+=gYvK
+-----END PGP MESSAGE-----`
+
+const e2eTestPayload = `{
+	"critical": {
+        "identity": {
+            "docker-reference": "gcr.io/kritis-project/kritis-server"
+        },
+        "image": {
+            "docker-manifest-digest": "sha256:b0be9acc0c8ca214f64fa4e0e214abc0bf342e8a4bc23b7e842ab21cfb50a6ea"
+        },
+    "type": "Google cloud binauthz container signature"
+    }
+}`
+
+const testImage = "gcr.io/kritis-project/kritis-server@sha256:b0be9acc0c8ca214f64fa4e0e214abc0bf342e8a4bc23b7e842ab21cfb50a6ea"
+
+func TestEndToEndWithAttestations(t *testing.T) {
+	signer, err := NewPgpSigner([]byte(e2ePrivateKey))
+	if err != nil {
+		t.Fatalf("Error initializing signer: %v", err)
+	}
+
+	att, err := signer.CreateAttestation([]byte(e2eTestPayload))
+	if err != nil {
+		t.Fatalf("Error creating attestation: %v", err)
+	}
+
+	publicKey := NewPublicKey(Pgp, []byte(e2ePublicKey), e2eKeyID)
```

Comment:
Outside scope for this PR: I believe we haven't add key id check for pgp keys. Can you create an issue for this?